pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -608,9 +639,11 @@ func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 
 	out := xdshttpfault.HTTPFault{}
 	if in.Delay != nil {
-		out.Delay = &xdsfault.FaultDelay{
-			Type:    xdsfault.FaultDelay_FIXED,
-			Percent: uint32(in.Delay.Percent),
+		out.Delay = &xdsfault.FaultDelay{Type: xdsfault.FaultDelay_FIXED}
+		if in.Delay.Percentage != nil {
+			out.Delay.Percentage = translatePercentToFractionalPercent(in.Delay.Percentage)
+		} else {
+			out.Delay.Percentage = translateIntegerToFractionalPercent(in.Delay.Percent)
```

Comment:
Look around in this file for code that checks the istio proxy version and generates config based on it. Please follow similar logic for this. 

But you now have to add a is11Proxy function instead of using old is10proxy